### DISCUSSION TITLE
Products not appearing in DNN Search

### DISCUSSION THREAD

**Discussion by MatthewJRyan** [openstore-ecommerce/OpenStore#165] [Q&A] [answered]
Currently my products are not appearing in the DNN search results so i would like to diagnose this.
The search results modules settings shows OS_Product View content type is ticked 
other module content is being indexed ok

But i dont know where to go from here to diagnose why products dont appear.

Can you give me a few hints on places to look, like database tables that should be populated, or is it done based on the sitemap? or Where is it done in code so i can start to look there?

**Comment by DNNMonster** [comment]
Did you find this?

https://github.com/nbrightproject/NBrightBuyDnnIdx

**Comment by MatthewJRyan** [comment]
it installed without error but It appears there is nothing in the installer zip in the source at that link
as no dll is added to the bin folder and nothing appears in the plugins list once installed

**Reply by MatthewJRyan** [reply]
there is also no NBS scheuler present in the DNN scheudler

ill try and create from the docs,

**Comment by DNNMonster** [comment]
There is info on the scheduler set up which you probably already found here: https://doc.openstore-ecommerce.com/

The NBrightBuyDnnIdx project will need to have it's net framework updated but otherwise it should compile.  

Here is a screen shot of a functional plugin configuration.  It's pretty cool once it's working for you and you won't likely need to look at it again.

![idx_plugin](https://user-images.githubusercontent.com/12777553/124838939-af7a6d00-df3c-11eb-8520-5f6a580e4c94.png)

**Reply by MatthewJRyan** [reply]
Thanks.
I had to do a fair bit to get it working for me as im using OpenUrlRewriter and DNN 9.9.1
But the good news is that i have;

- DNN 9.9.1 with 
- OpenStore V4.1.4 - custom templates and bo
- NBrightBuyMigrate - Migrated data from really old nbstore
- NBrightBuyDnnIdx - Indexing working with openurl and now includes meta tags
- OpenUrlRewriter - had to fix quite alot here including AddURL sp

Thanks for all you help, super useful.
If you want my changes let me know ill find a way to get them to you

**Comment by DNNMonster** [comment]
You powered through it.  Glad it's working for you.  I have a few customizations going on in the Indexer to enable role based displays.  If you have code that fixes or improves things then I'm sure everyone would love to see them.  You might start with an issue so that others can give you some feedback.  Dave's the long time maintainer so he will let you know if a particular issue is a good candidate for a pull request.  He can also stop you from spending time if he knows it won't make it into the code base.

**Comment by leedavi** [comment]
@Matthew,  Sorry for the late reply, for some reason Outlook insists in putting github email in my spam folder. Despite telling it not to.

What code have you changed?

**Reply by matt-binary-synergy** [reply]
Looks like most of it wont be of interest as its not in OpenStore itself (or at least not something you want to include like templates etc). 

The only ones you "_might_" care about are the stores OpenUrlRewriter provider and the NBrightBuyDnnIdx search indexer.

As i said im not exactly familiar with using github correctly to ive just forked the relevant repos, branched from the revision we started making changes from and pushed it back to my fork as a 9.9.1 branch. Feel free to do whatever from there to determine if any of it is useful to you. Its all [here](https://github.com/binary-synergy)

**Comment by MatthewJRyan** [comment]
I will see if i can get some time to do a diff on a freshly downloaded source but im currently busy trying to get the OS_DisplaySearch module to work (or more acturatly, trying to learn how it works as im sure it does).

**Comment by DNNMonster** [comment]
Most of the js that is dealing with the search is happening here:

https://github.com/openstore-ecommerce/OpenStore/blob/ce09324695cc817ceb49919fa0dfc902950595ea/Themes/ClassicAjax/Default/Search.cshtml

There is click and keyup enter event both of which are responsible for setting hidden input.  If either of the handlers fire and generate the hidden input it will in turn trigger the jquery change handler on the .searchreturn div.  

That change handler essentially does a page reload but with the NavigationData in place which will get processed server side.

Most of the interesting server side logic happens at about line 1817 here in the Get Filter Setup region:

https://github.com/openstore-ecommerce/OpenStore/blob/4f03e8f3977639ec9f13e5356f9b53c02074eef5/Components/Product/ProductFunctions.cs#L1817

**Comment by leedavi** [comment, marked as answer]
The index module has been updated to run on v4.

https://github.com/openstore-ecommerce/OS_DnnIdx
